lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp

```diff
@@ -2790,7 +2759,7 @@ void SymbolFileDWARF::FindTypes(const TypeQuery &query, TypeResults &results) {
   // With -gsimple-template-names, a templated type's DW_AT_name will not
   // contain the template parameters. Try again stripping '<' and anything
   // after, filtering out entries with template parameters that don't match.
-  if (!have_index_match) {
+  if (!have_index_match && !query.GetSearchByMangledName()) {
```

Comment:
This `!query.GetSearchByMangledName()` shouldn't be necessary since this condition is also in line 2730. so `have_index_match` will always be true

Reply:
Actually, never mind, if no type is found `have_index_match` will be false even when searching by mangled name, so this condition is necessary.